cnc_ctrl_v1/CNC_Functions.h

```diff
@@ -1192,6 +1192,111 @@ void  setSpindlePower(boolean powerState) {
     maslowDelay(delayAfterChange);
 }
 
+void PIDTestVelocity(Axis* axis, const float start, const float stop, const float steps){
+    // Moves the defined Axis at series of speed steps for PID tuning
+    // Start Log
+    Serial.println(F("--PID Velocity Test Start--"));
+    Serial.println("Axis=" + axis->motorGearboxEncoder.name());
+    Serial.println(axis->motorGearboxEncoder.getPIDString());
+
+    double startTime;
+    double print = micros();
+    double current = micros();
+    float lastPosition = axis->motorGearboxEncoder.encoder.read();
+    float error;
+    float distMoved;
+    float reportedSpeed;
+    float span = stop - start;
+    float speed;
+    
+    // Start the steps
+    axis->disablePositionPID();
+    axis->attach();
+    for(int i = 0; i < steps; i++){
+        // 1 step = start, 2 step = start & finish, 3 = start, start + 1/2 span...
+        speed = start;
+        if (i > 0){
+            speed = start + (span * (i/(steps-1)));
+        }
+        startTime = micros();
+        axis->motorGearboxEncoder.write(speed);
+        while (startTime + 2000000 > current){
+          if (current - print > 20000){
+            // Calculate and log error on same frequency as PID interrupt
+            distMoved   =  axis->motorGearboxEncoder.encoder.read() - lastPosition;
+            reportedSpeed= (7364.0*distMoved)/float(current - print);  //6*10^7 us per minute, 8148 steps per revolution
```

Comment:
This 7364 was replaced with a calculated value recently that might be better to use, not a big issue